`get_store_from_url` fails due to missing import `redis`

Without `redis` installed, we currently get

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
src/quantcore/thek/utils.py:698: in inner_f
    return f(*args, **kwargs)
src/quantcore/thek/view.py:134: in __init__
    if self.exists and partition_on is None:
src/quantcore/thek/kartothek_dataset.py:107: in exists
    return DatasetMetadata.exists(uuid=self.uuid, store=self.store())
/opt/conda/envs/quantcore.thek/lib/python3.7/site-packages/minimalkv/_get_store.py:55: in get_store_from_url
    return get_store(**url2dict(url))
/opt/conda/envs/quantcore.thek/lib/python3.7/site-packages/minimalkv/_get_store.py:145: in get_store
    store = create_store(type, params)
/opt/conda/envs/quantcore.thek/lib/python3.7/site-packages/minimalkv/_store_creation.py:23: in create_store
    return _create_store_hfs(type, params)
/opt/conda/envs/quantcore.thek/lib/python3.7/site-packages/minimalkv/_store_creation.py:114: in _create_store_hfs
    from minimalkv._hstores import HFilesystemStore
/opt/conda/envs/quantcore.thek/lib/python3.7/site-packages/minimalkv/_hstores.py:6: in <module>
    from minimalkv.memory.redisstore import RedisStore
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
    import re
    from io import BytesIO
    from typing import IO, Iterator, List, Optional, Union
>   from redis import StrictRedis
E   ModuleNotFoundError: No module named 'redis'
```

when running `get_store_from_url` in an environment without `redis-py` installed. This is not surprising, as we don't put the import into the class / function scopes as we e.g. do for the `GoogleCloudStore`. Similarly for the `GitCommitStore`. The question here IMO is how this worked before.

--- a/minimalkv/memory/redisstore.py
+++ b/minimalkv/memory/redisstore.py
@@ -2,11 +2,14 @@
 # -*- coding: utf-8 -*-
 import re
 from io import BytesIO
-from typing import IO, Iterator, List, Optional, Union
+from typing import IO, TYPE_CHECKING, Iterator, List, Optional, Union
 
-from redis import StrictRedis
+if TYPE_CHECKING:
+    from redis import StrictRedis
 
-from .. import FOREVER, NOT_SET, KeyValueStore, TimeToLiveMixin
+from minimalkv._constants import FOREVER, NOT_SET
+from minimalkv._key_value_store import KeyValueStore
+from minimalkv._mixins import TimeToLiveMixin
 
 
 class RedisStore(TimeToLiveMixin, KeyValueStore):
@@ -19,7 +22,7 @@ class RedisStore(TimeToLiveMixin, KeyValueStore):
 
     """
 
-    def __init__(self, redis: StrictRedis):
+    def __init__(self, redis: "StrictRedis"):
         self.redis = redis
 
     def _delete(self, key: str) -> int:

--- a/tests/test_import.py
+++ b/tests/test_import.py
@@ -1,0 +1,19 @@
+import mock
+
+optional_dependencies = ["redis", "dulwich", "sqlalchemy", "google", "azure", "boto"]
+
+orig_import = __import__
+
+
+def mock_optional_dependencies(name, *args):
+    if name in optional_dependencies:
+        raise ImportError(f"Cannot import optional dependency {name}.")
+
+    return orig_import(name, *args)
+
+
+def test_import():
+    with mock.patch("builtins.__import__", side_effect=mock_optional_dependencies):
+        from minimalkv import get_store_from_url
+
+        get_store_from_url("hfs:///tmp")

